### PR TITLE
Feat: support preventing component from garbage collection

### PR DESCRIPTION
**Pull request by chivalryq**
### Description of your changes

Now we only support prevent trait-generated resources from garbage collecion.
This PR add support for this feature on component

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by Somefive — APPROVED** [review]
Good Job!

File: apis/core.oam.dev/v1alpha1/garbagecollect_types.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ type GarbageCollectPolicyRule struct {
 // GarbageCollectPolicyRuleSelector select the targets of the rule
 type GarbageCollectPolicyRuleSelector struct {
 	TraitTypes []string `json:"traitTypes"`
+	CompTypes  []string `json:"compTypes"`
```
Comment (Somefive) [inline apis/core.oam.dev/v1alpha1/garbagecollect_types.go:50]: Add some comments here. If compTypes and traitTypes are both used, the combination logic is `or`.

Comment (wonderflow) [inline apis/core.oam.dev/v1alpha1/garbagecollect_types.go:50]:
```suggestion
	CompTypes  []string `json:"componentTypes"`
```

Comment (chivalryq) [reply apis/core.oam.dev/v1alpha1/garbagecollect_types.go]: done

Comment (chivalryq) [reply apis/core.oam.dev/v1alpha1/garbagecollect_types.go]: done

**Review by wonderflow — APPROVED** [review]

**Pull request merged**